radix-engine/src/model/nodes/bucket.rs

```diff
@@ -293,37 +296,30 @@ impl Bucket {
                 let _: ConsumingBucketBurnInput = scrypto_decode(&args.raw)
                     .map_err(|e| InvokeError::Error(BucketError::InvalidRequestData(e)))?;
 
-                let bucket: Bucket = system_api
-                    .node_drop(&node_id)
-                    .map_err(InvokeError::Downstream)?
-                    .into();
-
-                // Notify resource manager, TODO: Should not need to notify manually
-                let resource_address = bucket.resource_address();
-                let mut value = system_api
-                    .borrow_node_mut(&RENodeId::ResourceManager(resource_address))
+                let node_ref = system_api
+                    .borrow_node(&node_id)
```

Comment:
Why just borrow here? Is this a change of abstraction?

Reply:
This is because when invoking `Bucket::burn`, the call frame only needs to **read** the resource address, such that it can call `ResourceManager::burn` by passing the `self`.